.github/workflows/all_solutions.yml

```
@@ -33,13 +33,43 @@ env:
   DOTNET_NOLOGO: true
 
 jobs:
+  check-modified-files:
+    name: Check if source files were modified, skip remaining jobs if not
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    # don't run this job if triggered by Dependabot, will cause all other jobs to be skipped as well
+    # don't run if triggered by a release or manual execution, as we're going to do the full build in those cases anyway
+    if: ${{ github.actor != 'dependabot[bot]' }} || !(github.event.release || github.event_name == 'workflow_dispatch')
```

I don't think we need the braces and they also don't cover the whole statement.

```suggestion
    if: github.actor != 'dependabot[bot]' || !(github.event.release || github.event_name == 'workflow_dispatch')
```